dex/elrond_dex_farm/src/lib.rs

```diff
@@ -368,35 +363,33 @@ pub trait Farm:
     ) -> SCResult<()> {
         let reward_token_id = self.reward_token_id().get();
         require!(token_in == reward_token_id, "Bad fee token identifier");
-        self.increase_actual_reserves(&amount);
+        require!(amount > 0, "Zero amount in");
+        self.increase_temporary_fee_storage(&amount);
         Ok(())
     }
 
     #[view(calculateRewardsForGivenPosition)]
     fn calculate_rewards_for_given_position(
         &self,
-        liquidity: Self::BigUint,
+        amount: Self::BigUint,
         attributes_raw: BoxedBytes,
     ) -> SCResult<Self::BigUint> {
-        require!(liquidity > 0, "Zero liquidity input");
+        require!(amount > 0, "Zero liquidity input");
         let farm_token_supply = self.farm_token_supply().get();
-        require!(farm_token_supply > liquidity, "Not enough supply");
+        require!(farm_token_supply >= amount, "Not enough supply");
 
-        let attributes = self.decode_attributes(&attributes_raw)?;
-        require!(
-            liquidity <= attributes.total_liquidity_amount,
-            "Bad arguments"
-        );
+        let current_block = self.blockchain().get_block_nonce();
+        let to_be_minted = self.calculate_per_block_rewards(current_block);
```

Comment:
in case of farm we need a vector of nonce / rewards change as we might change the rewards per block multiple times.

Reply:
from excel examples this is not needed. as the rewards are actually given from rewards reserve. On update reward rate we do generate rewards and update price per share.

Reply:
done